### PR TITLE
Make TensorFlow DALI operator able to return the arbitral number of o…

### PR DESCRIPTION
**Pull request by JanuszL**
…utputs

- add C API to return the number of available pipeline outputs
- reworks TensorFlow DALI operator to return a list of tensors instead
  of a fixed number of 2

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

File: dali/c_api/c_api.h [hunk 1]
```diff
@@ -69,6 +69,10 @@ extern "C" {
    */
   DLL_PUBLIC int64_t* daliShapeAt(daliPipelineHandle* pipe_handle, int n);
 
+  /**
+   * @brief Returns number of DALI pipeline outputs
+   */
```
Comment (Kh4L) [inline dali/c_api/c_api.h]: We could return an `unsigned` here

File: dali/tensorflow/daliop.cc [hunk 2]
```diff
@@ -60,20 +61,20 @@ tf::TensorShape DaliToShape(int64_t* ns) {
...
-    tf::PartialTensorShape shape;
-    TF_RETURN_IF_ERROR(c->GetAttr("shape", &shape));
+    // define shape for the first output
+    std::vector<tf::PartialTensorShape> shapes;
+    TF_RETURN_IF_ERROR(c->GetAttr("shapes", &shapes));
     tf::shape_inference::ShapeHandle passed_shape;
     TF_RETURN_IF_ERROR(
-        c->MakeShapeFromPartialTensorShape(shape, &passed_shape));
+        c->MakeShapeFromPartialTensorShape(shapes[0], &passed_shape));
     TF_RETURN_IF_ERROR(
```
Comment (Kh4L) [inline dali/tensorflow/daliop.cc]: Static shape inference should be done for each ouput:
```
for (unsigned i = 0; i < shapes.size(); ++i) {
    tf::shape_inference::ShapeHandle passed_shape;
    TF_RETURN_IF_ERROR(
        c->MakeShapeFromPartialTensorShape(shapes[i], &passed_shape));
     c->set_output(i, passed_shape); 
}
```

Comment (Kh4L) [reply dali/tensorflow/daliop.cc]: And we should remove 
https://github.com/NVIDIA/DALI/blob/dea4f1a1249cbdcc3e5f0fd200ddd184ad5aad5d/dali/tensorflow/daliop.cc#L60

**Comment by JanuszL** [comment]
I need to update examples as well.

File: dali/tensorflow/daliop.cc (hunk 2 above)
Comment (JanuszL) [reply dali/tensorflow/daliop.cc]: Done

File: dali/c_api/c_api.h (hunk 1 above)
Comment (JanuszL) [reply dali/c_api/c_api.h]: Done

**Comment by JanuszL** [comment]
Another build 38613.

**Comment by JanuszL** [comment]
Build after resolving conflicts 38631.

**Pull request merged**